### PR TITLE
Remove fetching spiffe bundles from endpoints for default template

### PR DESCRIPTION
**Pull request by shankgan**
For Control plane Authentication, Istiod can authenticate the sidecar/gateways using JWT. Requiring Istiod to authenticate the dataplane using the mTLS cert signed by the CA is unnecessary.

### CODE REVIEW

**Review by zerobfd — APPROVED** [review]
LGTM, will merge when a domain expert takes a look.

**Comment by howardjohn** [comment]
Checking something, please don't merge yet

**Comment by zerobfd** [comment]
Just for posterity: there was an internal convo and we in fact didn't merge until the checking was done.

**Pull request merged**
